conda/plugins/__init__.py

```diff
@@ -23,14 +23,11 @@ class CondaSubcommand(NamedTuple):
 
     :param name: Subcommand name (e.g., ``conda my-subcommand-name``).
     :param summary: Subcommand summary, will be shown in ``conda --help``.
-    :param action: Callable that will be run when the subcommand is invoked.
+    :param add_argument_parser: Function that adds the parser to the main conda argument parser
     """
     name: str
     summary: str
-    action: Callable[
-        [List[str]],  # arguments
-        Optional[int],  # return code
-    ]
+    add_argument_parser: Callable
```

Comment:
FWIW @FFY00 and me considered this pattern in the past and erred on the side of NOT making the `argparse` API accessible to end users, to prevent wild growth since `argparse` is arguable not a great API and we wouldn't want to encourage following anti-patterns. Supporting click etc was specifically not a design goal.

I'd suggest to look into whether we can renaming the existing design a `BasicCondaSubcommand` and adding a `ArgparseCondaSubcommand` class to handle the `argparse` specific APIs.